src/lib/bootloader/proposal_client.rb

```diff
@@ -7,6 +7,38 @@
 module Bootloader
   # Proposal client for bootloader configuration
   class ProposalClient < ::Installation::ProposalClient
+    # Error when during update media is booted by different technology than target system.
+    class MismatchBootloader < RuntimeError
+      include Yast::I18n
+
+      def initialize(old_bootloader, new_bootloader)
+        @old_bootloader = old_bootloader
+        @new_bootloader = new_bootloader
+
+        super("Mismatching bootloaders")
+      end
+
+      def user_message
+        textdomain "bootloader"
+
+        boot_map = {
+          # TRANSLATORS: kind of boot. It is term for way how x86_64 can boot
+          "grub2"     => _("Legacy BIOS boot"),
+          # TRANSLATORS: kind of boot. It is term for way how x86_64 can boot
+          "grub2-efi" => _("EFI boot")
+        }
+        # TRANSLATORS: keep %{} intact. It will be replaced by kind of boot
+        format(_(
+                 "Cannot update bootloader as there is mismatch of boot technology.<br>" \
```

Comment:
Better wording + some nice rich text formatting to make it better readable:
```ruby
"<p>Cannot upgrade the bootloader because of a mismatch of the boot technology. " \
"The upgraded system uses <i>%{old_boot}</i> while the installation medium " \
"has been booted using <i>%{new_boot}</i>.</p>" \
"<p>This scenario is not supported, the upgraded system may not boot " \
"or the upgrade process can fail later.</p>"
```